AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift

```diff
@@ -0,0 +1,293 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+#if canImport(AdyenAuthentication)
+    @_spi(AdyenInternal) import Adyen
+    import Adyen3DS2
+    import AdyenAuthentication
+    import Foundation
+    import UIKit
+
+    internal enum DelegateAuthenticationError: LocalizedError {
+        case registrationFailed(cause: Error?)
+        case authenticationFailed(cause: Error?)
+    
+        internal var errorDescription: String? {
+            switch self {
+            case let .registrationFailed(causeError):
+                if let causeError = causeError {
+                    return "Registration failure caused by error: { \(causeError.localizedDescription) }"
+                } else {
+                    return "Registration failure."
+                }
+            case let .authenticationFailed(causeError):
+                if let causeError = causeError {
+                    return "Authentication failure caused by error: { \(causeError.localizedDescription) }"
+                } else {
+                    return "Authentication failure."
+                }
+            }
+        }
+    }
+
+    /// Handles the 3D Secure 2 fingerprint and challenge actions separately + Delegated Authentication.
+    @available(iOS 14.0, *)
+    internal class ThreeDS2PlusDACoreActionHandler: AnyThreeDS2CoreActionHandler {
```

Comment:
I believe `ThreeDS2PlusDACoreActionHandler` and `ThreeDS2CoreActionHandler` have plenty in common. Maybe we can inherit one from another?

Reply:
the differences make it very difficult to inherit one from the other and reuse code between them, because the differences sort of make the common code invalid for the other one.

and if it was possible, having them separate is cleaner IMHO

Reply:
its very difficult to unentangle the common code and the different code, I think its cleaner to just separate them, and as a sacrifice we'll have a little duplication unfortunately

Reply:
it toke me a while but I made `ThreeDS2PlusDACoreActionHandler ` inherit from `ThreeDS2CoreActionHandler`, Thanks for pointing this out :-)